functions/http/2.0/index.js

```diff
@@ -0,0 +1,45 @@
+import templayed from './templayed';
+
+const parseHeaders = (headers) =>
+  headers.reduce((acc, { key, value }) => ({ ...acc, [key]: value }), {});
+
+const parseQueryParameters = (queryParameters) =>
+  queryParameters
+    .map(({ key, value }, index) => {
+      const paramKey = index === 0 ? `?${key}` : key;
+      return `${paramKey}=${encodeURIComponent(value)}`;
+    })
+    .join('&');
+
+const generateUrl = (url, protocol, queryParameters) =>
+  `${protocol}://${url}${parseQueryParameters(queryParameters)}`;
+
+const http = async ({
+  url,
+  method,
+  body,
+  headers = [],
+  protocol,
+  variables,
+  queryParameters = [],
+}) => {
+  const variableMap = variables.reduce((previousValue, currentValue) => {
+    previousValue[currentValue.key] = currentValue.value;
+    return previousValue;
+  }, {});
+  
+
+  const fetchUrl = generateUrl(url, protocol, queryParameters);
+  const options = {
+    method,
+    headers: parseHeaders(headers),
+    body: body ? templayed(body)(variableMap) : undefined,
```

Comment:
maybe use null here instead of undefined: https://stackoverflow.com/questions/5076944/what-is-the-difference-between-null-and-undefined-in-javascript

Reply:
In case of a GET request, `body` needs to be `undefined` or else it will send an empty body along with the GET-request resulting in an error